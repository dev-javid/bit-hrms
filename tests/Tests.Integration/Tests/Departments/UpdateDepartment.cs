namespace Tests.Integration.Tests.Departments
{
    public class UpdateDepartment(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/departments/1";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Departments/UpdateDepartment.sql");
            var command = new
            {
                Name = Faker.Random.String(3, 50)
            };
            HttpResponseMessage response;
            using (StringContent content = new StringContent(JsonSerializer.Serialize(command), Encoding.UTF8, JsoneMediaType))
            {
                response = await Client.PutAsync(Route, content);
            }

            var result = await response.Content.ReadAsStringAsync();
            response.Dispose();
            Approvals.VerifyJson(result.Ignore("$.name"));
        }
    }
}
