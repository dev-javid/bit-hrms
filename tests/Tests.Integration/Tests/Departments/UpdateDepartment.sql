INSERT INTO public.departments (id, company_id, name, created_by, created_on)
VALUES
(1, 999, 'HR', 1, now());