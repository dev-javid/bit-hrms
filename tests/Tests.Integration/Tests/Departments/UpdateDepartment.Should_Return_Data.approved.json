{
    "departmentId": 1,
    "name": "__IGNORED_VALUE__",
    "jobTitles": []
}